Darkly/LDUserBuilder.m

```diff
@@ -228,8 +235,22 @@ -(User *)build
         [user setAnonymous:currentAnonymous];
     }
     DEBUG_LOG(@"LDUserBuilder building User with anonymous: %d", anonymous);
-    [[DataManager sharedManager] saveContext];
+    [self saveUser: user];
     return user;
 }
 
+-(void) saveUser: (User *) user {
+    [[DataManager sharedManager] purgeOldUsers];
+    UserEntity *userEntity = [[DataManager sharedManager] findUserEntityWithkey:user.key];
+    
+    if (userEntity) {
```

Comment:
Do we need to update the users updatedAt value if its found from CoreData? Currently we are only updating in the User init method when we create a new object.